### PR TITLE
Integrated open library API

### PR DESCRIPTION
**Pull request by storresy**
**Summary**
Added open library service
Added endpoint to get book info using book isbn

**Trello Card**
https://trello.com/c/Wg8x5Xkq

**Screenshots**
screen shot 2018-12-18 at 11 28 39

### CODE REVIEW

File: app/services/open_library_service.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class OpenLibraryService
+  include HTTParty
+  base_uri 'https://openlibrary.org'
+
+  def initialize(format = 'json', jscmd = 'data')
+    @options = { query: { format: format, jscmd: jscmd } }
+  end
+
+  def book_info(identifier, identifier_type = 'ISBN')
+    @options[:query][:bibkeys] = "#{identifier_type}:#{identifier}"
```
Comment (storresy) [inline app/services/open_library_service.rb:10]: I added the identifier type optional parameter to allow searching books using other types of identifiers

File: app/services/open_library_service.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  def request_book_info
+    self.class.get('/api/books', @options)
+  end
+
+  def parsed_book_info(api_book_info)
+    book_info = api_book_info[@options[:query][:bibkeys]]
+    return {} if api_book_info.blank?
+
+    {
+      isbn: book_info['identifiers']['isbn_10'].first,
```
Comment (Gmarn) [inline app/services/open_library_service.rb:25]: Here you can using directly the one you sent, can't you?

File: app/services/open_library_service.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      title: book_info['title'],
+      subtitle: book_info['subtitle'],
+      number_of_pages: book_info['number_of_pages'],
+      authors: book_info['authors'].map { |a| a['name'] }
```
Comment (Gmarn) [inline app/services/open_library_service.rb:29]: Do not forgot to use explicit name: `a` -> `author`

File: app/controllers/api/v1/books_controller.rb [hunk 4]
```diff
@@ -9,11 +9,20 @@ def show
         render json: book
       end
 
+      def info
+        book_info = OpenLibraryService.new.book_info(isbn_param)
+        render json: book_info, status: book_info.present? ? :ok : :not_found
```
Comment (Gmarn) [inline app/controllers/api/v1/books_controller.rb:14]: Maybe we can have better handle of error, to know when is a service error or only a `not_found`. Just to think ;)

**Review by Gmarn — APPROVED** [review]

**Pull request merged**